### ISSUE TITLE
exe file generated by PyInstaller not running

### ISSUE THREAD

**Issue by saubiswa** [pyinstaller/pyinstaller#2752]
I am getting the following error while running exe file generated by PyInstaller. I am using Windows 10. Please help.

**File "site-packages\ruamel\yaml\reader.py", line 100, in stream
AttributeError: 'SafeLoader' object has no attribute '_stream'**
                         ------details deleted-----
During handling of the above exception, another exception occurred:
                        ------details deleted-----
**File "site-packages\ruamel\yaml\reader.py", line 102, in stream
ruamel.yaml.error.YAMLStreamError: input stream needs to specified**

I have followed the steps in
•"If Things Go Wrong" Guide
 (https://github.com/pyinstaller/pyinstaller/wiki/If-Things-Go-Wrong) and
•make sure everything is packaged correctly
https://github.com/pyinstaller/pyinstaller/wiki/How-to-Report-Bugs#make-sure-everything-is-packaged-correctly
but the problem remains.
I have also used the latest development version with no effect.

Detailed log file is attached for your reference.
[Log.txt](https://github.com/pyinstaller/pyinstaller/files/1236639/Log.txt)

**Comment by htgoebel** [comment]
This looks like not all required files are packages. Please have a **deep** look at [Make sure everything is packaged correctly](https://github.com/pyinstaller/pyinstaller/wiki/How-to-Report-Bugs#make-sure-everything-is-packaged-correctly) for tips to find out what is missing. Also have a look at the source of ruamel\yaml\reader.py.

**Comment by saubiswa** [comment]
I double checked, doesn't look like any file is missing or I couldn't catch it.
I also updated ruamel\yaml\reader.py with the latest version from Git.
The problem still remains.

**Comment by htgoebel** [comment]
Please provide a *minimal* example to reproduce the error and the *exact* versions of all used software.

**Comment by saubiswa** [comment]
I am using dateparser to parse datetime from string, in my program.
Installed dateparser (v- 0.6.0) using pip (v- 9.0.1), ruamel (v- 0.15.31) was installed along with as pre-requisite.
Program runs fine thru PyCharm (v- 2017.2.1).
Creating exe using pyinstaller (v - 3.3.dev0+f0df2d2bb). Log available in the previously attached log file.
The exe not running in the same environment. Log available in the previously attached log file.

**Comment by htgoebel** [comment]
Still missing a minimal example.

Did you tray you program without PyCharm, on the console?

**Comment by htgoebel** [comment]
For reference:
```
[13420] LOADER: Running Auth24x7.py
Traceback (most recent call last):
  File "site-packages\ruamel\yaml\reader.py", line 100, in stream
AttributeError: 'SafeLoader' object has no attribute '_stream'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "Auth24x7.py", line 11, in <module>
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 673, in _load_unlocked
  File "c:\users\saubiswa\appdata\local\programs\python\python35-32\lib\site-packages\PyInstaller\loader\pyimod03_importers.py", line 631, in exec_module
    exec(bytecode, module.__dict__)
  File "site-packages\dateparser\__init__.py", line 4, in <module>
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 673, in _load_unlocked
  File "c:\users\saubiswa\appdata\local\programs\python\python35-32\lib\site-packages\PyInstaller\loader\pyimod03_importers.py", line 631, in exec_module
    exec(bytecode, module.__dict__)
  File "site-packages\dateparser\date.py", line 11, in <module>
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 673, in _load_unlocked
  File "c:\users\saubiswa\appdata\local\programs\python\python35-32\lib\site-packages\PyInstaller\loader\pyimod03_importers.py", line 631, in exec_module
    exec(bytecode, module.__dict__)
  File "site-packages\dateparser\date_parser.py", line 10, in <module>
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 673, in _load_unlocked
  File "c:\users\saubiswa\appdata\local\programs\python\python35-32\lib\site-packages\PyInstaller\loader\pyimod03_importers.py", line 631, in exec_module
    exec(bytecode, module.__dict__)
  File "site-packages\dateparser\conf.py", line 66, in <module>
  File "site-packages\dateparser\conf.py", line 32, in __init__
  File "site-packages\dateparser\conf.py", line 46, in _get_settings_from_yaml
  File "site-packages\dateparser\utils\__init__.py", line 188, in __init__
  File "site-packages\ruamel\yaml\loader.py", line 36, in __init__
  File "site-packages\ruamel\yaml\scanner.py", line 83, in __init__
  File "site-packages\ruamel\yaml\scanner.py", line 99, in reset_scanner
  File "site-packages\ruamel\yaml\scanner.py", line 424, in fetch_stream_start
  File "site-packages\ruamel\yaml\reader.py", line 159, in get_mark
  File "site-packages\ruamel\yaml\reader.py", line 102, in stream
ruamel.yaml.error.YAMLStreamError: input stream needs to specified
[13420] Failed to execute script Auth24x7
```

**Comment by kyeer** [comment]
Was this resolved at all? Having same issue.

**Comment by saubiswa** [comment]
no it wasn't.

**Comment by htgoebel** [comment]
Nobody provided an example, https://github.com/pyinstaller/pyinstaller/issues/2752#issuecomment-323937942 was not answered.

**Comment by zachanator070** [comment]
I was able to reproduce by creating example.py:
`import dateparser
print(dateparser.parse("today")`
then
`pyinstaller example.py`
which produces:
`33 INFO: PyInstaller: 3.3
33 INFO: Python: 3.5.3
33 INFO: Platform: Linux-4.10.0-37-generic-x86_64-with-Ubuntu-17.04-zesty
34 INFO: wrote /home/zach/git/pyinstaller_example/example.spec
36 INFO: UPX is not available.
37 INFO: Extending PYTHONPATH with paths
['/home/zach/git/pyinstaller_example', '/home/zach/git/pyinstaller_example']
37 INFO: checking Analysis
37 INFO: Building Analysis because out00-Analysis.toc is non existent
37 INFO: Initializing module dependency graph...
38 INFO: Initializing module graph hooks...
42 INFO: Analyzing base_library.zip ...
3065 INFO: running Analysis out00-Analysis.toc
3101 INFO: Caching module hooks...
3105 INFO: Analyzing /home/zach/git/pyinstaller_example/example.py
3698 INFO: Processing pre-find module path hook   distutils
5082 INFO: Processing pre-safe import module hook   six.moves
5454 INFO: Loading module hooks...
5454 INFO: Loading module hook "hook-lib2to3.py"...
5456 INFO: Loading module hook "hook-pkg_resources.py"...
5993 INFO: Processing pre-safe import module hook   win32com
5996 INFO: Loading module hook "hook-sysconfig.py"...
5997 INFO: Loading module hook "hook-pydoc.py"...
5998 INFO: Loading module hook "hook-distutils.py"...
5999 INFO: Loading module hook "hook-xml.py"...
6231 INFO: Loading module hook "hook-encodings.py"...
6287 INFO: Loading module hook "hook-pytz.py"...
6302 INFO: Loading module hook "hook-regex.py"...
6317 INFO: Looking for ctypes DLLs
6380 WARNING: library user32 required via ctypes not found
6415 INFO: Analyzing run-time hooks ...
6420 INFO: Including run-time hook 'pyi_rth_pkgres.py'
6421 INFO: Including run-time hook 'pyi_rth_multiprocessing.py'
6436 INFO: Looking for dynamic libraries
6730 INFO: Looking for eggs
6731 INFO: Python library not in binary dependencies. Doing additional searching...
6780 INFO: Using Python library /usr/lib/x86_64-linux-gnu/libpython3.5m.so.1.0
6784 INFO: Warnings written to /home/zach/git/pyinstaller_example/build/example/warnexample.txt
6822 INFO: Graph cross-reference written to /home/zach/git/pyinstaller_example/build/example/xref-example.html
6858 INFO: checking PYZ
6858 INFO: Building PYZ because out00-PYZ.toc is non existent
6859 INFO: Building PYZ (ZlibArchive) /home/zach/git/pyinstaller_example/build/example/out00-PYZ.pyz
7293 INFO: Building PYZ (ZlibArchive) /home/zach/git/pyinstaller_example/build/example/out00-PYZ.pyz completed successfully.
7297 INFO: checking PKG
7297 INFO: Building PKG because out00-PKG.toc is non existent
7297 INFO: Building PKG (CArchive) out00-PKG.pkg
7340 INFO: Building PKG (CArchive) out00-PKG.pkg completed successfully.
7341 INFO: Bootloader /usr/local/lib/python3.5/dist-packages/PyInstaller/bootloader/Linux-64bit/run
7341 INFO: checking EXE
7341 INFO: Building EXE because out00-EXE.toc is non existent
7341 INFO: Building EXE from out00-EXE.toc
7342 INFO: Appending archive to ELF section in EXE /home/zach/git/pyinstaller_example/build/example/example
7359 INFO: Building EXE from out00-EXE.toc completed successfully.
7362 INFO: checking COLLECT
7362 INFO: Building COLLECT because out00-COLLECT.toc is non existent
7362 INFO: Building COLLECT out00-COLLECT.toc
7724 INFO: Building COLLECT out00-COLLECT.toc completed successfully.`

The directory looks like this after creating the binary:
`zach@bartok:~/git/pyinstaller_example $ ls
build  dist  example.py  example.spec  __pycache__
zach@bartok:~/git/pyinstaller_example $ ls dist/example
_bz2.so               _codecs_kr.so         _hashlib.so           libbz2.so.1.0         libpython3.5m.so.1.0  _lzma.so              pytz/                 _ssl.so               
_codecs_cn.so         _codecs_tw.so         include/              libcrypto.so.1.0.0    libreadline.so.7      mmap.so               readline.so           termios.so            
_codecs_hk.so         _ctypes.so            _json.so              libexpat.so.1         libssl.so.1.0.0       _multibytecodec.so    _regex.so             
_codecs_iso2022.so    _decimal.so           lib/                  liblzma.so.5          libtinfo.so.5         _multiprocessing.so   resource.so           
_codecs_jp.so         example               lib2to3/              libmpdec.so.2         libz.so.1             _opcode.so            _ruamel_yaml.so `

Then running the executable gives me this:
`$ ./dist/example/example 
Traceback (most recent call last):
  File "ruamel/yaml/reader.py", line 100, in stream
AttributeError: 'SafeLoader' object has no attribute '_stream'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "example.py", line 1, in <module>
    import dateparser
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 673, in _load_unlocked
  File "/usr/local/lib/python3.5/dist-packages/PyInstaller/loader/pyimod03_importers.py", line 631, in exec_module
    exec(bytecode, module.__dict__)
  File "dateparser/__init__.py", line 4, in <module>
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 673, in _load_unlocked
  File "/usr/local/lib/python3.5/dist-packages/PyInstaller/loader/pyimod03_importers.py", line 631, in exec_module
    exec(bytecode, module.__dict__)
  File "dateparser/date.py", line 11, in <module>
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 673, in _load_unlocked
  File "/usr/local/lib/python3.5/dist-packages/PyInstaller/loader/pyimod03_importers.py", line 631, in exec_module
    exec(bytecode, module.__dict__)
  File "dateparser/date_parser.py", line 10, in <module>
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 673, in _load_unlocked
  File "/usr/local/lib/python3.5/dist-packages/PyInstaller/loader/pyimod03_importers.py", line 631, in exec_module
    exec(bytecode, module.__dict__)
  File "dateparser/conf.py", line 66, in <module>
  File "dateparser/conf.py", line 32, in __init__
  File "dateparser/conf.py", line 46, in _get_settings_from_yaml
  File "dateparser/utils/__init__.py", line 188, in __init__
  File "ruamel/yaml/loader.py", line 36, in __init__
  File "ruamel/yaml/scanner.py", line 83, in __init__
  File "ruamel/yaml/scanner.py", line 99, in reset_scanner
  File "ruamel/yaml/scanner.py", line 424, in fetch_stream_start
  File "ruamel/yaml/reader.py", line 159, in get_mark
  File "ruamel/yaml/reader.py", line 102, in stream
ruamel.yaml.error.YAMLStreamError: input stream needs to specified
[117016] Failed to execute script example`

Info about dateparser:
`Name: dateparser
Version: 0.6.0
Summary: Date parsing library designed to parse dates from HTML pages
Home-page: https://github.com/scrapinghub/dateparser
Author: Scrapinghub
Author-email: info@scrapinghub.com
License: BSD
Location: /usr/local/lib/python3.5/dist-packages
Requires: ruamel.yaml, python-dateutil, regex, pytz, tzlocal
`

Info about ruamel.yaml I am using:
`
$ pip show ruamel.yaml
Name: ruamel.yaml
Version: 0.15.34
Summary: ruamel.yaml is a YAML parser/emitter that supports roundtrip preservation of comments, seq/map flow style, and map key order
Home-page: https://bitbucket.org/ruamel/yaml
Author: Anthon van der Neut
Author-email: a.van.der.neut@ruamel.eu
License: MIT license
Location: /usr/local/lib/python3.5/dist-packages
Requires:`

Would you like an example using the ruamel.yaml library?

**Comment by zachanator070** [comment]
I found the problem to be with this bit of code in the dateparser package. I picked out the bits that I think relate to this issue and produce the same error that we have been seeing:
```
from pkgutil import get_data
import ruamel.yaml
import ruamel.yaml.loader
import ruamel.yaml.reader

class SafeLoader(ruamel.yaml.loader.SafeLoader):
    def __init__(self, *args, **kwds):
       super(SafeLoader, self).__init__(*args, **kwds)

data = get_data('data', 'settings.yaml')
SafeLoader(data)
```

The output when running this snippet:

```
$ ./dist/example/example 
Traceback (most recent call last):
  File "ruamel/yaml/reader.py", line 100, in stream
AttributeError: 'SafeLoader' object has no attribute '_stream'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "example.py", line 11, in <module>
    SafeLoader(data)
  File "example.py", line 8, in __init__
    super(SafeLoader, self).__init__(*args, **kwds)
  File "ruamel/yaml/loader.py", line 36, in __init__
  File "ruamel/yaml/scanner.py", line 83, in __init__
  File "ruamel/yaml/scanner.py", line 99, in reset_scanner
  File "ruamel/yaml/scanner.py", line 424, in fetch_stream_start
  File "ruamel/yaml/reader.py", line 159, in get_mark
  File "ruamel/yaml/reader.py", line 102, in stream
ruamel.yaml.error.YAMLStreamError: input stream needs to specified
[7761] Failed to execute script example
```

**Comment by Lynten** [comment]
Having the same issue with the `dateparser` package.

I solved the problem through adding the `data` directory and the `_strptime.py` module to the `dist/example` dir after building the project.

1. `data`: the directory under `C:\Programs\Python27\Lib\site-packages`, which contains YAML setting files `dateparser` use. ( See [`dateparser/conf.py` line 45](https://github.com/scrapinghub/dateparser/blob/1c54cea710bee5b239d68ed0ef102873a5192b8c/dateparser/conf.py#L45) )
2. `_strptime.py`: the Python module under `C:\Programs\Python27\Lib`, which is the module `dateparser` dynamically import. ( See  [`dateparser/utils/strptime.py` line 26](https://github.com/scrapinghub/dateparser/blob/1c54cea710bee5b239d68ed0ef102873a5192b8c/dateparser/utils/strptime.py#L26) )

The structure:
* dist
  * example
    * data
    * _strptime.py

**Comment by rockstar8577** [comment]
Just doing something as simple as this has issue. If i comment out the import it's fine.
```python
import dateparser

print ("Hello")
```

**Comment by lbcg** [comment]
I found a fix that could work on other machines:
Since it's not imported by hooks, you need to ask pyinstaller to put `Lib\_strptime.py` into the directory of the script.
In the .spec file used in `pyinstaller mypythonfile.spec`, that is automatically generated by running the command `pyinstaller mypythonfile.py`

You can change the **data** parameter, to grab **_strptime**
`
import  imp  # allows for finding the path of the module
# example for my own application
a = Analysis(['.\\mypythonfile.py'],  # 
             pathex=['.\\pyinstaller'],
             binaries=[],
             datas= [(imp.find_module('_strptime')[1], '.')]  # here is the important part
             ...
`

**Comment by htgoebel** [comment]
Should already be fixed in #3790
